python/rikai/contrib/s3.py

```diff
@@ -0,0 +1,49 @@
+#  Copyright (c) 2021 Rikai Authors
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Various s3 utilities for Rikai functionality. For example, libraries like
+OpenCV cannot open s3 url's directly. Instead we make a presigned s3 url.
+"""
+import logging
+from urllib.parse import urlparse
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError
+except ImportError as e:
+    raise ImportError("Please `pip install awscli` to use s3 utils") from e
```

Comment:
should we put `boto3` into `rikai[aws]` package? and  here , why did we ask to install `awscli` while it is `boto3` missing?

Reply:
i remember you removed that aws extras package for a specific purpose? If we can add it back that makes it easier

Reply:
yea, i removed it because i found that we can use `pyarrow` for s3 access.  If you need `boto3` here, lets just add them